### ISSUE TITLE
NETCore 3.1 System.Diagnostics.Tracing file not found running on docker image

### ISSUE THREAD

**Issue by gonzaloluna** [dotnet/dotnet-docker#2741]
I'm trying to trace network details for my API using EventListener from System.Diagnostics.Tracing.
Things work fine on Windows however when running the project on a docker image I'm getting the following error :

_**Event Microsoft-System-Net-Primitives - EventSourceMessage : ERROR: Exception in Command Processing for EventSource Microsoft-System-Net-Primitives: Could not load file or assembly 'System.Diagnostics.Tracing, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.**_

 I tried this in the WeatherForecast sample project and I'm getting the same issue, so this does not seem to be a references conflict in my project.
Here is my dockerfile. The project has no nugets at all. 
Installing the System.Diagnostics.Tracing package didn't work either
<PackageReference Include="System.Diagnostics.Tracing" Version="4.3.0" />
```
FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
WORKDIR /app
ADD . .

RUN dotnet restore -s https://api.nuget.org/v3/index.json

RUN dotnet publish \
  --self-contained true \
  --runtime linux-x64 \
  /p:PublishTrimmed=true \
  -c Release \
  -o ./output
RUN chmod +x -R ./output/

FROM mcr.microsoft.com/dotnet/core/aspnet:3.1
RUN adduser \
  --disabled-password \
  --home /app \
  --gecos '' app \
  && chown -R app /app

USER app
WORKDIR /app
COPY --from=build-env /app/output .

ENV DOTNET_RUNNING_IN_CONTAINER=true \
  ASPNETCORE_URLS=http://+:8080
EXPOSE 8080
ENTRYPOINT ["./WebApplication1", "--urls", "http://0.0.0.0:8080"]
```

Thanks!

**Comment by mthalman** [comment]
I wonder if this is related to the `/p:PublishTrimmed=true` option.  Have you tried without that?

Could you provide an example project that repros this behavior?

**Comment by gonzaloluna** [comment]
I tried so many things I got stuck but /p:PublishTrimmed=true did work! Thanks!

**Comment by mthalman** [comment]
Just to clarify, are you saying that after removing `/p:PublishTrimmed=true` caused your application to work?

I'd still like to determine a repro of this.  I've tried my own app but have not been able to reproduce this behavior.  Can you provide an example project?

**Comment by gonzaloluna** [comment]
That's correct, removing /p:PublishTrimmed=true did work in both, the sample project and production project.
Here is the sample project. https://github.com/gonzaloluna/NetworkEventSourceLoggerTest

**Comment by mthalman** [comment]
I tried the sample project at https://github.com/gonzaloluna/NetworkEventSourceLoggerTest and it worked fine for me even when including `/p:PublishTrimmed=true`.

Here are the steps I took:
1. `git clone https://github.com/gonzaloluna/NetworkEventSourceLoggerTest`
2. `cd NetworkEventSourceLoggerTest`
3. Modify Dockerfile to include `/p:PublishTrimmed=true`:

```Dockerfile
RUN dotnet publish \
  /p:PublishTrimmed=true \
  --self-contained true \
  --runtime linux-x64 \
  -c Release \
  -o ./output
```

4. `docker build --pull -t app .`
5. `docker run --rm -it -p 8000:8080 app`
6. Open localhost:8000/WeatherForecast in my browser.  Got successful results.
